### DISCUSSION TITLE
[HW5] How much freedom do we have for the algorithms

### DISCUSSION THREAD

**Discussion by JustinKim98** [prosyslab-classroom/cs424-program-reasoning#453] [Q&A] [answered]
### Your Name

김재우

### Question

I tried to implement the HW5 algorithm based on what we learned from the lecture note. However, to enhance the performance, I'm willing to modify this algorithm with another approach.
However, it may not be compatible with learn - intersect - pick algorithm that we learned in the class.
I'm curious if we are allowed to change this.

HW5알고리즘을 수업 시간에 배운대로 구현했지만, 이보다 성능을 높이기 위해 다른 방식의 알고리즘을 사용해도 되는지 궁금합니다. 수업에서 나온 대로 learn - intersect - pick의 순서를 유지하는것이 필수적인가요?, 아니면 거기서 벗어나서 다른 알고리즘을 사용해도 되나요? (learn-intersect-pick함수의 signature를 바꾸지 않는다는 가정 하에)

(이전에 올려주신 Duet 논문도 읽어보았지만, 수업 시간에 배운 순서와는 조금 다르게 작동하는 것으로 보입니다)

**Comment by KihongHeo** [comment, marked as answer]
OK. Give it a try.
